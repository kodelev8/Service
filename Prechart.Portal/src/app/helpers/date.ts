import { Injectable } from "@angular/core";
import * as moment from "moment";

@Injectable({ providedIn: 'root' })
export class DateHelper {

  formatDate(date: any) {
    return moment(date).local().format('yyyy-MM-DD');
  }

}
