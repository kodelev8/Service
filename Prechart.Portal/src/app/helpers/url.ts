import { Injectable } from "@angular/core";
import { environment } from "src/environments/environment";

@Injectable({ providedIn: 'root' })
export class UrlHelper {
  format(subUrl: string, path: string) {
    var getUrl = this.getUrl(subUrl);

    if (environment.production) {
      var host = environment.apiUrl.replace('[[subdomain]]', getUrl);
      return `${host}${path}`;
    }
    else {
      var host = environment.apiUrl.replace('[[port]]', getUrl);
      return `${host}${path}`;
    }
  }

  getUrl(domain: string) {
    var value: string = '';
    switch (domain) {
      case 'document': value = environment.document;
        break;
      case 'belastingen': value = environment.belastingen;
        break;
      case 'user': value = environment.user;
        break;
      case 'loonheffings': value = environment.loonheffings;
        break;
      case 'person': value = environment.person;
        break;
      case 'werkgever': value = environment.werkgever;
        break;
      default: value = '0000';
        break;
    }
    return value;
  }
}
