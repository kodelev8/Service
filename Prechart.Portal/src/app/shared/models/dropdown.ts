export class Dropdown {
  public value: number;
  public label: string;
}
