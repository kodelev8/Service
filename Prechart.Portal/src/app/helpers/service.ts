import { HttpClient, HttpHeaders } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { AccountService } from "../shared/account/services/account.service";
import { UrlHelper } from "./url";
import {User} from "../shared/account/models/user";

@Injectable({ providedIn: 'root' })
export class ServiceHelper {
  headers: HttpHeaders;
  constructor(
    private http: HttpClient,
    private accountService: AccountService,
    private urlHelper: UrlHelper)
  {
    if (this.accountService.userValue) {
      var token = this.accountService.userValue;

      this.headers = new HttpHeaders({
        'Authorization': `Bearer ${token}`,
      });
    }
  }

  get(port: string, path: string) {
    var url = this.urlHelper.format(port, path);
    return this.http.get<any>(url, { headers: this.headers });
  }

  post(port: string, path: string, object: any) {
    var url = this.urlHelper.format(port, path);
    return this.http.post<any>(url, object, { headers: this.headers });
  }

  validateUser(port: string, path: string, object: any, options: any) {
    var url = this.urlHelper.format(port, path);
    return this.http.post<any>(url, object, options);
  }
}
