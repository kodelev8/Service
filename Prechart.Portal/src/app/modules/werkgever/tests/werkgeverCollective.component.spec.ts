import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { FormsModule } from '@angular/forms';
import { BrowserModule, By } from '@angular/platform-browser';
import { HttpClientModule } from '@angular/common/http';
import { RouterTestingModule } from '@angular/router/testing';

import { WerkgeverCollectiveComponent } from 'src/app/modules/werkgever/collective/werkgeverCollective.component';
import { ServiceHelper } from '../../../helpers/service';
import { AccountService } from '../../../shared/account/services/account.service';
import { PanelModule } from 'primeng/panel';
import { AccordionModule } from 'primeng/accordion';
import { DropdownModule } from 'primeng/dropdown';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

describe('WerkgeverCollectiveComponent', () => {
  let fixture: ComponentFixture<WerkgeverCollectiveComponent>;
  let component: WerkgeverCollectiveComponent;
  let element: any;

  const getMockFn = jest.fn();
  let service: ServiceHelper;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        HttpClientTestingModule,
        HttpClientModule,
        RouterTestingModule,
        BrowserAnimationsModule,
        FormsModule,
        PanelModule,
        AccordionModule,
        DropdownModule,
      ],
      declarations: [
        WerkgeverCollectiveComponent,
      ],
      providers: [AccountService],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(WerkgeverCollectiveComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    fixture.detectChanges();
    service = TestBed.inject(ServiceHelper);

    service.get = getMockFn;
  });

  it('should create the Werkgever Collective page', () => {
    expect(component).toBeTruthy();
  });

  it('should render the Werkgever Collective page correctly', () => {
    expect(element).toMatchSnapshot();
  });

  //it('should have the Werkgever Management table', () => {
  //  const wrapper = fixture.debugElement.query(By.css(".search"));
  //  expect(wrapper).toBeTruthy();
  //  expect(element).toMatchSnapshot();
  //});

});
