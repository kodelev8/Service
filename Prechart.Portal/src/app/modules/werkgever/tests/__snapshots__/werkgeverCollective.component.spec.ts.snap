// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WerkgeverCollectiveComponent should render the Werkgever Collective page correctly 1`] = `
<div
  id="root1"
>
  <div
    class="div-center p-2"
  >
    <h3>
      Werkgever - Collectieve
    </h3>
    <p-dropdown
      class="p-element p-inputwrapper ng-untouched ng-pristine ng-valid"
      optionlabel="label"
      placeholder="Selecteer een werkgever"
      styleclass="m-2"
    >
      <div
        class="m-2 p-dropdown p-component"
      >
        <div
          class="p-hidden-accessible"
        >
          <input
            aria-expanded="false"
            aria-haspopup="listbox"
            placeholder="Selecteer een werkgever"
            readonly=""
            role="combobox"
            type="text"
          />
        </div>
        
        <span
          class="p-dropdown-label p-inputtext p-placeholder"
        >
          Selecteer een werkgever
        </span>
        
        
        
        <div
          aria-haspopup="listbox"
          aria-label="dropdown trigger"
          class="p-dropdown-trigger"
          role="button"
        >
          <span
            class="p-dropdown-trigger-icon pi pi-chevron-down"
          />
        </div>
        
      </div>
    </p-dropdown>
    <p-panel
      class="p-element m-1 text-left"
      header="Werkgevergegevens"
    >
      <div
        class="p-panel p-component"
        id="p-panel-2"
      >
        <div
          class="p-panel-header"
          id="p-panel-2-titlebar"
        >
          <span
            class="p-panel-title"
            id="p-panel-2_header"
          >
            Werkgevergegevens
          </span>
          
          
          <div
            class="p-panel-icons p-panel-icons-end"
            role="tablist"
          >
            
            
          </div>
        </div>
        
        <div
          aria-hidden="false"
          aria-labelledby="p-panel-2-titlebar"
          class="p-toggleable-content ng-trigger ng-trigger-panelContent"
          id="p-panel-2-content"
          role="region"
          style=""
        >
          <div
            class="p-panel-content"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-6"
                  >
                    <strong>
                      Naam
                    </strong>
                  </div>
                  <div
                    class="col-3"
                  >
                    <span>
                      
                    </span>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-6"
                  >
                    <strong>
                      Sector
                    </strong>
                  </div>
                  <div
                    class="col-3"
                  >
                    <span>
                      
                    </span>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-6"
                  >
                    <strong>
                      Fiscaalnummer
                    </strong>
                  </div>
                  <div
                    class="col-3"
                  >
                    <span>
                      
                    </span>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-6"
                  >
                    <strong>
                      Klantnaam
                    </strong>
                  </div>
                  <div
                    class="col-3"
                  >
                    <span>
                      
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="col"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-6"
                  >
                    <strong>
                      Loonheffingen Extentie
                    </strong>
                  </div>
                  <div
                    class="col-3"
                  >
                    <span>
                      
                    </span>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-6"
                  >
                    <strong>
                      Omzetbelasting Extentie
                    </strong>
                  </div>
                  <div
                    class="col-3"
                  >
                    <span>
                      
                    </span>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-6"
                  >
                    <strong>
                      Datum Actief Vanaf
                    </strong>
                  </div>
                  <div
                    class="col-3"
                  >
                    <span>
                      
                    </span>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-6"
                  >
                    <strong>
                      Datum Actief Tot
                    </strong>
                  </div>
                  <div
                    class="col-3"
                  >
                    <span>
                      
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="text-right"
            >
              
            </div>
            
          </div>
          
        </div>
      </div>
    </p-panel>
    <p-panel
      class="p-element m-1 text-left"
      header="Collectieve Aangifte"
    >
      <div
        class="p-panel p-component"
        id="p-panel-3"
      >
        <div
          class="p-panel-header"
          id="p-panel-3-titlebar"
        >
          <span
            class="p-panel-title"
            id="p-panel-3_header"
          >
            Collectieve Aangifte
          </span>
          
          
          <div
            class="p-panel-icons p-panel-icons-end"
            role="tablist"
          >
            
            
          </div>
        </div>
        
        <div
          aria-hidden="false"
          aria-labelledby="p-panel-3-titlebar"
          class="p-toggleable-content ng-trigger ng-trigger-panelContent"
          id="p-panel-3-content"
          role="region"
          style=""
        >
          <div
            class="p-panel-content"
          >
            <p-accordion
              class="p-element"
            >
              <div
                class="p-accordion p-component"
                role="tablist"
              >
                
              </div>
            </p-accordion>
            
          </div>
          
        </div>
      </div>
    </p-panel>
  </div>
</div>
`;
